### ISSUE TITLE
Android Test App Dependencies are Out Of Date specifcally no.nordicsemi.android:dfu:1.8.1., Unable to Build

### ISSUE THREAD

**Issue by z3ugma** [megahealth/TestBleLib#7]
This repo is trying to implement no.nordicsemi.android:dfu:1.8.1. version 1.8.1 (that doesn't exist anymore) and there's a disclaimer on their site saying that they are no longer maintaining old versions https://github.com/NordicSemiconductor/Android-DFU-Library#library
The libraries exist, but those specific versions are not available to download. 
**Basically we would be forced to upgrade those libraries to the latest versions and migrate the code to get it working, since they can no longer be downloaded from the vendor** 

We are attempting to take advantage of the latest commit with `Enso` mode.

We want an APK that we can build off of this commit 
https://github.com/megahealth/TestBleLib/commit/d43260a76869671d12ad622ec0eef6aa6e51ce1b#diff-8a15be1f8b25fdd3e3a0061a06476bf8a0563f8cf65b4d18f9eb2ddb9cdf6ee9

Noting that `SimpleMainActivity.kt` has our enso mode button (a button with all lowercase enable enso mode)

```

                        <Button
                            android:id="@+id/btn_enable_enso_mode"
                            android:layout_width="wrap_content"
                            android:layout_height="wrap_content"
                            android:text="enable enso mode"
                            android:textColor="@color/colorAccent" />
```

While attempting to build the demo app, we were able to upgrade gradle, but the project is quite outdated. 
I was able to get gradle running using  7.2
./gradlew assembleDebug
distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-bin.zip

Several dependencies are no longer available, including
```
com.leon:lfilepickerlibrary:1.8.0
 no.nordicsemi.android:dfu:1.8.1  (from Nordic)
 com.google.android:flexbox
```
The Nordic library for Bluetooth control seems to require a significant migration based on their official documentation. 

Should I proceed with updating the project to make it functional, or should I escalate this to Megahealth?
Here is the output after trying to build

```
> Could not resolve all task dependencies for configuration ':app:debugCompileClasspath'.
   > Could not find com.leon:lfilepickerlibrary:1.8.0.
     Required by:
         project :app
   > Could not find no.nordicsemi.android:dfu:1.8.1.
     Required by:
         project :app
   > Could not find com.google.android:flexbox:0.3.2.
     Required by:
         project :app
```

This repo is trying to implement  version 1.8.1 (that doesn't exist anymore) and there's a disclaimer on their site saying that they are no longer maintaining old versions https://github.com/NordicSemiconductor/Android-DFU-Library#library
The libraries exist, but those specific versions are not available to download. 
Basically we would be forced to upgrade those libraries to the latest versions and migrate the code to get it working, since they can no longer be downloaded from the vendor

**Comment by z3ugma** [comment]
Maybe this was fixed with https://github.com/megahealth/TestBleLib/commit/9beed1df0d076333ec86b105d8633a4c04a5e1de ?

**Comment by z3ugma** [comment]
Speciicall in that commit: 

```
//    implementation 'no.nordicsemi.android:dfu:1.8.1'
    implementation 'no.nordicsemi.android:dfu:2.0.2'

    ```
